contrib/mesos/pkg/scheduler/podtask/predicate.go

```diff
@@ -57,21 +61,29 @@ func NodeSelectorPredicate(t *T, offer *mesos.Offer) bool {
 
 	// check the NodeSelector
 	if len(t.Pod.Spec.NodeSelector) > 0 {
-		slaveLabels := map[string]string{}
-		for _, a := range offer.Attributes {
-			if a.GetType() == mesos.Value_TEXT {
-				slaveLabels[a.GetName()] = a.GetText().GetValue()
+		l := map[string]string{
+			"kubernetes.io/hostname": offer.GetHostname(),
+		}
+		if n != nil && n.Labels != nil {
+			for k, v := range n.Labels {
+				if !strings.HasPrefix(k, node.LabelPrefix) {
```

Comment:
I will move that into the node module. But we have to filter for those slave labels first. It might be that the node had old slave labels. If we don't filter, the old ones will still apply, i.e. the scheduler will see labels which are not set anymore.